### DISCUSSION TITLE
Custom value option in combobox

### DISCUSSION THREAD

**Discussion by kveperedo** [adobe/react-spectrum#7235] [Q&A] [answered]
Hello! I'm building a combobox component that allows a custom value which is supported out of the box. Is it possible to display a selectable custom value option instead of showing the empty state? Thank you!

![2024-10-23 at 09 11 AM@2x](https://github.com/user-attachments/assets/9a6e2549-5cd4-432b-ac6b-41d079fbb755)

**Comment by snowystinger** [comment, marked as answer]
I think this should be doable already. You'd create an item which always matches the filter. When that key is selected, you'd add the item to your more permanent set/state with some new/different key and immediately select it.

You'd need to control the selection state.

**Reply by kveperedo** [reply]
Thanks for the quick response! Is there an example wherein the selection state is controlled?

**Reply by snowystinger** [reply]
https://react-spectrum.adobe.com/react-aria/ComboBox.html#selection
